module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,67 @@
+#include "compression.hpp"
+
+#include <iostream>
+#include <string>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    std::array<std::array<uint8_t, height>, width>& uncompressed) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(height * width);
+
+    for (const auto& row : uncompressed) {
+        auto rowIterator = row.cbegin();
+        uint8_t value = *rowIterator;
+        uint8_t currentValueCounter = 1;
+        rowIterator++;
+
+        while (rowIterator != row.cend()) {
+            if (*rowIterator == value) {
+                currentValueCounter++;
+            } else {
+                compressed.push_back(std::make_pair(value, currentValueCounter));
+                value = *rowIterator;
+                currentValueCounter = 1;
+            }
+            rowIterator++;
+        }
+        // and for last value in a row:
+        compressed.push_back(std::make_pair(value, currentValueCounter));
+    }
+
+    compressed.shrink_to_fit();
+    return compressed;
+}
+
+std::array<std::array<uint8_t, height>, width> decompressGrayscale(
+    std::vector<std::pair<uint8_t, uint8_t>>& compressed) {
+    std::array<std::array<uint8_t, height>, width> uncompressed;
+    auto pixelGroup = compressed.begin();
+    uint8_t howManyPixels = pixelGroup->second;
```

Comment:
Not needed. You have this value inside pixelGroup->second;. So copy is redundant